src/app/ReadHandler.cpp

```diff
@@ -61,6 +62,54 @@ ReadHandler::ReadHandler(ManagementCallback & apCallback, Messaging::ExchangeCon
     mSessionHandle.Grab(mExchangeCtx->GetSessionHandle());
 }
 
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+ReadHandler::ReadHandler(ManagementCallback & apCallback) :
+    mExchangeCtx(*this), mManagementCallback(apCallback), mOnConnectedCallback(HandleDeviceConnected, this),
+    mOnConnectionFailureCallback(HandleDeviceConnectionFailure, this)
+{
+    mInteractionType = InteractionType::Subscribe;
+    mFlags.ClearAll();
+}
+
+void ReadHandler::ResumeSubscription(CASESessionManager * caseSessionManager,
+                                     SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo)
+{
+    mSubscriptionId          = subscriptionInfo.mSubscriptionId;
+    mMinIntervalFloorSeconds = subscriptionInfo.mMinInterval;
+    mMaxInterval             = subscriptionInfo.mMaxInterval;
+    SetStateFlag(ReadHandlerFlags::FabricFiltered, subscriptionInfo.mFabricFiltered);
+
+    // Move dynamically allocated attributes and events from the SubscriptionInfo struct into the object pool managed by the IM
+    // engine
+    CHIP_ERROR err;
+    for (size_t i = 0; i < subscriptionInfo.mAttributePaths.AllocatedCount(); i++)
+    {
+        AttributePathParams attributePathParams = subscriptionInfo.mAttributePaths[i].GetParams();
+        err = InteractionModelEngine::GetInstance()->PushFrontAttributePathList(mpAttributePathList, attributePathParams);
```

Comment:
Overall, this forces the subscriptions to appear that everything is dirty. If an ICD had gone to sleep for 10 minutes, and is waking up due to a delta of just a single attribute, this will try to fulfill ALL the paths, even if just that one item has changed in the interim, and the device knows it. This makes all the reports significantly larger than they need to be. Since the stack remembers what subscriptions it had, the product could be made to mark the subscriptions as dirty. Overall, it seems that this whole scheme would scale better if there was a way to provide, in `Server` init, a policy object for marking what has changed since last time a report was done, which only a specific product can tell. This, however, could only be done with changes to how dirty paths are marked. 
It may be that because the method here is fire-and-forget, there is no better way than full priming, since otherwise the paths would need to be marked dirty after all the secure sessions have been re-initiated.

Reply:
Filed issue #24534